hub/core/tensor.py

```diff
@@ -168,8 +188,7 @@ def _check_array_and_tensor_are_compatible(tensor_meta: dict, array: np.ndarray)
             "max_shape", tensor_meta["max_shape"], len(sample_shape)
         )
 
-    # TODO: remove these once dynamic shapes are supported and update docstring
-    if not np.array_equal(tensor_meta["max_shape"], sample_shape):
-        raise NotImplementedError("Dynamic shapes are not supported yet.")
-    if not np.array_equal(tensor_meta["min_shape"], sample_shape):
-        raise NotImplementedError("Dynamic shapes are not supported yet.")
+
+def _update_tensor_meta_shapes(shape: Tuple[int], tensor_meta: dict):
```

Comment:
this will be handled by a dedicated meta class (@thisiseshan will be building that)